project/src/main/java/com/google/sps/data/GeoContext.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.maps.GeoApiContext;
+
+/**
+ * GeoContext holds the entry point for a Google GEO API request.
+ * It is initiated once and used for all Google GEO EPI requests,
+ * including Google Places API and Google Distance Matrix API.
+ *
+ */
+
+public final class GeoContext {
+
+    private static GeoContext singleInstance = null;
+
+    private GeoApiContext context;
+
+    private GeoContext() {
+        context = new GeoApiContext.Builder()
+            .apiKey(System.getenv("API_KEY"))
+            .build();
+    }
+
+    /**
+     * Creates a GeoContext instance if not yet been anitialized and returns the
+     * intance's contex. The double check of the instance being null is for thread safety.
+     * @return a GeoApiContext object used for Google GEO API requests
+     */
+    public static GeoApiContext getContext() {
+        if (singleInstance == null) {
+            synchronized (GeoContext.class) {
+                if (singleInstance == null) {
+                    singleInstance = new GeoContext();
+                }
+            }
+        }
+        return singleInstance.context;
```

Comment:
could this fail in earlier stages? there is no way this will try to get the "context" of a null object? or an error will be thrown if an earlier stage fails?

Reply:
That's a good point. This shouldn't fail because if there is any problem with the API key the GeoApiContext will still be built with no errors. The errors will come only when the context is used by a request. Actualy I found out that this case (of invalid API key) wasn't treated in the fetcher so I added it there.